src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,160 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index, size + 1);
+        if (index == 0 && size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(getNodeByIndex(index).prev, value, getNodeByIndex(index));
+            getNodeByIndex(index).prev.next = newNode;
+            getNodeByIndex(index).prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        }
+        if (size == list.size()) {
+            head = tail = (Node<T>) list.get(0);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Objects.checkIndex(index, size);
+        if (index == 0) {
+            return head.value;
+        } else if (index == size - 1) {
+            return tail.value;
+        } else {
```

Comment:
You already check it in your GetNodeByIndex method.

Don’t create repeating code.
If the logic of your code repeats - move it to the separate private method. Remember about DRY and KISS principles.